docs/configuration/hyperparameter_optimization.md

```diff
@@ -30,145 +33,86 @@ hyperopt:
 
-- `space`: Use `linear` or `log`.
-- `range.low`: the minimum value the parameter can have
-- `range.high`: the maximum value the parameter can have
-- `steps`: (optional) number of steps to break down a range.
+For numeric `spaces`, these define the range where the value is generated
 
-For instance `range: (0.0, 1.0), steps: 3` would yield `[0.0, 0.5, 1.0]` as potential values to sample from, while if
-`steps` is not specified, the full range between `0.0` and `1.0` will be used.
+- `lower`: the minimum value the parameter can have
+- `upper`: the maximum value the parameter can have
+- `q`: quantization number, used in `spaces` such as `quniform`, `qloguniform`, `qrandn`, `qrandint`, `qlograndint`
+- `base`: defines the base of the log for `loguniform`, `qloguniform`, `lograndint` and `qlograndint`
 
-Float example:
+**Float example**: Uniform floating point random values (in log space) between 0.001 and 0.1
 
 ```yaml
 trainer.learning_rate:
-  space: linear
-  range:
-    low: 0.001
-    high: 0.1
-  steps: 4
+  space: loguniform
+  lower: 0.001
+  uppper: 0.1
 ```
 
-Integer example:
+**Integer example**: Uniform random integer values 1, 2, 3
 
 ```yaml
 combiner.num_fc_layers:
-  space: linear
-  range:
-    low: 1
-    high: 4
-```
-
-## Categorical selection
-
-- `space`: Use `choice`.
-- `categories`: a list of possible values. The type of each value of the list is not important (they could be strings,
-integers, floats and anything else, even entire dictionaries).
-
-Example:
-
-```yaml
-text.cell_type:
-  categories: [rnn, gru, lstm]
-  space: choice
+  space: randint
+  lower: 1
+  upper: 4
 ```
 
-# Sampler
-
-## Grid sampler
-
-The `grid` sampler creates a search space by exhaustively selecting all elements from the outer product of all possible
-combinations of hyperparameter values provided in the `parameters` section.
-
-To use `grid` sampling with `float` parameters, it is required to specify the number of `steps`.
-
-Example:
+**Quantized Example**: Uniform random floating point values such a 0, 0.1, 0.2, ..., 0.9
 
 ```yaml
-sampler:
-  type: grid
+my_output_feature.dropout:
+  space: quniform
+  lower: 0
+  upper: 1
+  q: 0.1
 ```
 
-## Random sampler
+## Categorical Hyperparameters
 
-The `random` sampler samples hyperparameter values randomly from the parameters search space.
-`num_samples` (default: `10`) can be specified in the `sampler` section.
+- `space`: Use `choice`.
+- `categories`: a list of possible values. The type of each value of the list is general, i.e., they could be strings,
+integers, floats and anything else, even entire dictionaries.  The values will be a uniform random selection.
 
 Example:
 
 ```yaml
-sampler:
-  type: random
-  num_samples: 10
+text.cell_type:
+  categories: [rnn, gru, lstm]
+  space: choice
```

Comment:
I sould always pu space first, just for consistency

Reply:
done.